concordium-std/src/lib.rs

```diff
@@ -273,9 +277,19 @@ pub use impls::*;
 pub use traits::*;
 pub use types::*;
 
+#[cfg(feature = "dl-malloc")]
+extern crate dlmalloc;
+// Use `globalDlmalloc` as the global allocator.
+#[allow(dead_code)]
+#[cfg(feature = "dl-malloc")]
+#[cfg_attr(not(feature = "std"), global_allocator)]
+static ALLOC: dlmalloc::GlobalDlmalloc = dlmalloc::GlobalDlmalloc;
+
+#[cfg(feature = "wee-alloc")]
 extern crate wee_alloc;
 // Use `wee_alloc` as the global allocator to reduce code size.
 #[global_allocator]
+#[cfg(feature = "wee-alloc")]
 static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
```

Comment:
I think this needs to be 
```suggestion
#[cfg_attr(feature = "wee-alloc", global_allocator)]
static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
```